src/docs/asciidoc/reference/concepts/zetapush-npm-modules.adoc

```diff
@@ -1,2 +1,69 @@
 :relative-path: ../../
 include::{docdir}/variables.adoc[]
+
+ZetaPush is a *monorepo* that contains several *npm modules*. Each of them have a specific role.
+
+== @zetapush/cli
+
+Contains the *Command Line Interface* of ZetaPush.
+This is thanks to this module that you can launch `zeta run` or `zeta push`. +
+You can have more information in his <<reference#cli, dedicated section>>.
+
+== @zetapush/client
+
+When you <<reference#client_types, develop the front part of your application>>, i.e. the web pages displayed to your users.
+You use in particular a *ZetaPush client* to interact with your worker and the *ZetaPush cloud*.
+
+== @zetapush/cometd
+
+You don't use this module directly.
+It is used internally for the communication with the *ZetaPush cloud* via the protocol link:{cometd-url}[CometD].
+
+== @zetapush/common
+
+This module comprises some functions or classes that are used in other modules.
+
+== @zetapush/core
+
+This is the core of *ZetaPush*. Notably, it contains the <<reference#dependency-injection, dependency injection>>
+
+== @zetapush/create
+
+When you launch `npm init @zetapush myApp` that this module which is used.
+It is responsible for create your *ZetaPush application* with the _CLI_.
+
+== @zetapush/example
+
+It contains a basic example of a *ZetaPush application*
+
+== @zetapush/http-server
+
+ZetaPush uses a HTTP server for some features, for example to confirm an account in the <<reference#integrate_user_management, Standard User Workflow>>. 
+You have information about this module in his <<reference#http_server, dedicated section>>
+
+== @zetapush/testing
+
+ZetaPush integrates testing and provides some util features about it via this module.
+
+== @zetapush/integration
+
+About testing, ZetaPush uses integration testing. Some util features are integrated in this module for this.
+
+== @zetapush/platform-legacy
+
+ZetaPush contains some _Technical Services_ that will be replaced step by step. 
+This module contains them and you have more information about their utilization in the <<reference#builtin_cloud_services, dedicated section>>.
+
+== @zetapush/troubleshooting
+
+ZetaPush helps for troubleshooting. In particular with the `zeta troubleshooting` command.
```

Comment:
When you have a technical error, it can be hard to understand what really happens. This module diagnoses all errors (errors when worker starts, errors when worker is pushed, errors in connection/network...) and gives you an help to understand the error and to help you fix it.

For example, a network error can have several causes:
* Your network card is disabled or misconfigured
* Your computer can't access the Internet
* Your computer is on a local network with a proxy but proxy is not reachable
* Your computer is on a local network with a proxy, proxy is reachable but the proxy can't access the Internet
* You have Internet access but ZetaPush cloud is not reachable
* ...

This module will give you help to understand where the problem is and how to fix it.